components/script/dom/htmlformcontrolscollection.rs

```diff
@@ -43,8 +47,40 @@ impl HTMLFormControlsCollection {
 
 impl HTMLFormControlsCollectionMethods for HTMLFormControlsCollection {
     // https://html.spec.whatwg.org/multipage/#dom-htmlformcontrolscollection-nameditem
-    fn NamedGetter(&self, name: DOMString, found: &mut bool) -> Option<Root<Element>> {
-        self.collection.NamedGetter(name, found)
+    fn NamedItem(&self, name: DOMString) -> Option<RadioNodeListOrElement> {
+        // Step 1
+        if name.is_empty() { return None; }
+
+        let count = self.collection.elements_iter().filter_map(|elem| {
+            if elem.get_string_attribute(&atom!("name")) == name
+               || elem.get_string_attribute(&atom!("id")) == name {
+                Some(elem)
+            } else { None }
```

Comment:
I would do that if the API is not unstable. Ideally, I would reuse this for the filter_map below, but .count() consumes the iterator so I had to repeat the entire mapping.